### PR TITLE
Exclude Epoch Boundary from Health Check

### PR DESCRIPTION
**Pull request by jedleggett**

### CODE REVIEW

File: relayer/src/health_manager.rs [hunk 1]
```diff
@@ -42,23 +42,28 @@ impl HealthManager {
...
+                                        datapoint_info!(
+                                            "relayer-health-state",
+                                            ("health_state", new_health_state, i64)
+                                        );
+                                }
                             }
                             recv(slot_receiver) -> maybe_slot => {
                                 let slot = maybe_slot.expect("error receiving slot, exiting");
+                                // Don't perform health updates within +/- ~30 sec of epoch boundary 
+                                outside_epoch_boundary = (75..=431_925).contains(&(slot % 432_000));
```
Comment (buffalu) [inline relayer/src/health_manager.rs:66]: epochs don't always line up like this. i think they do on mainnet, but testnet and local networks this isn't always the case.

Comment (segfaultdoc) [inline relayer/src/health_manager.rs:66]: nit use constant from solana-sdk crate:
`DEFAULT_SLOTS_PER_EPOCH`

File: relayer/src/health_manager.rs [hunk 2]
```diff
@@ -42,23 +42,28 @@ impl HealthManager {
...
+                                        *health_state.write().unwrap() = new_health_state;
+                                        datapoint_info!(
+                                            "relayer-health-state",
+                                            ("health_state", new_health_state, i64)
+                                        );
+                                }
                             }
                             recv(slot_receiver) -> maybe_slot => {
                                 let slot = maybe_slot.expect("error receiving slot, exiting");
+                                // Don't perform health updates within +/- ~30 sec of epoch boundary
```
Comment (buffalu) [inline relayer/src/health_manager.rs:65]: should use slot-based number here and not time

Comment (buffalu) [reply relayer/src/health_manager.rs:65]:
```
let slot_in_epoch = slot % SLOTS_PER_EPOCH; // (or read clock on-chain)
outside_epoch_boundary = (slot_in_epoch <= 75 || slot_in_epoch >= 431_925);
```

File: relayer/src/health_manager.rs (hunk 1 above)
Comment (segfaultdoc) [reply relayer/src/health_manager.rs:66]: can you just do
`75 < (slot % DEFAULT_SLOTS_PER_EPOCH) < 431_925`

Comment (jedleggett) [reply relayer/src/health_manager.rs:66]: clippy suggested this syntax

File: relayer/src/health_manager.rs [hunk 3]
```diff
@@ -42,23 +42,30 @@ impl HealthManager {
...
+                                            ("health_state", new_health_state, i64)
+                                        );
+                                }
                             }
                             recv(slot_receiver) -> maybe_slot => {
                                 let slot = maybe_slot.expect("error receiving slot, exiting");
+                                // Don't perform health updates within +/- 75 slots of epoch boundary
+                                // Note: This is not necessarily correct for local testing
+                                let slot_index = slot % DEFAULT_SLOTS_PER_EPOCH;
+                                outside_epoch_boundary = 75 < slot_index && slot_index < (DEFAULT_SLOTS_PER_EPOCH - 75);
```
Comment (esemeniuc) [inline relayer/src/health_manager.rs:68]: would be good to add a const for 75 with a comment on why the specific value

File: relayer/src/health_manager.rs [hunk 4]
```diff
@@ -42,23 +42,30 @@ impl HealthManager {
...
-                                    match last_update.elapsed() <= missing_slot_unhealthy_threshold {
-                                        true => HealthState::Healthy,
-                                        false => HealthState::Unhealthy,
-                                    };
-                                *health_state.write().unwrap() = new_health_state;
-                                datapoint_info!(
-                                    "relayer-health-state",
-                                    ("health_state", new_health_state, i64)
-                                );
+                                if outside_epoch_boundary {
```
Comment (buffalu) [inline relayer/src/health_manager.rs:50]: can you move the check of outside_epoch_boundary into the match statement/if/else for health state? may be helpful to continue emitting data points during that time

File: relayer/src/health_manager.rs [hunk 5]
```diff
@@ -42,23 +42,30 @@ impl HealthManager {
...
+                                        *health_state.write().unwrap() = new_health_state;
+                                        datapoint_info!(
+                                            "relayer-health-state",
+                                            ("health_state", new_health_state, i64)
+                                        );
+                                }
                             }
                             recv(slot_receiver) -> maybe_slot => {
                                 let slot = maybe_slot.expect("error receiving slot, exiting");
+                                // Don't perform health updates within +/- 75 slots of epoch boundary
```
Comment (buffalu) [inline relayer/src/health_manager.rs:65]: thoughts on moving this logic to the health state stuff above?

**Review by esemeniuc — APPROVED** [review]

**Pull request closed without merging**